src/renderer/components/BalanceGraph/BalanceGraph.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from "react";
+import {useState, useEffect} from "react";
+import { 
+    LineChart, Line, XAxis, Tooltip,
+} from "recharts";
+
+export enum IntervalEnum {
+    HOUR = "hour",
+    DAY = "day",
+    WEEK = "week",
+    MONTH = "month",
+    YEAR = "year",
+}
+enum Days {
+    SUN = 0,MON,TUE,WED,THU,FRI,SAT,
+}
+enum Months {
+    JAN = 0,FEB,MAR,APR,MAY,JUN,JUL,AUG,SEP,OCT,NOV,DEC
+}
+
+export interface IBalanceGraphProps {
+    defaultInterval?: IntervalEnum;
+    getData: () => Promise<number[]>;
+}
+
+export const BalanceGraph: React.FunctionComponent<IBalanceGraphProps> = (props: IBalanceGraphProps) => {
+    const [data, setData] = useState<Array<object>>([]);
+    const [interval, setInterval] = useState<IntervalEnum>(IntervalEnum.WEEK);
+
+    const formatDate = (date: string): string=>{
+        const month = date.slice(4,7).toUpperCase();
+        const day = date.slice(8,10);
+        return month + " " + day;
+    };
+
+    const setXAxis = (array: Promise<number[]>, time: IntervalEnum): void=>{
+        const dataArray: Array<object> = [];
+        const dateToday = new Date();
+        switch (time) {
+            case IntervalEnum.HOUR: {
+                const minute = dateToday.getMinutes();
+                for (let i = 5; i >=0; i--) {
+                    dataArray.push({
+                        name: ((minute-10*i)<0 ? 60+(minute-10*i) : (minute-10*i)) + "'",
+                        value: array[i]
+                    });
+                }
```

Comment:
separate each case as function in another file. like normalizeDataPoints(array, interval) and have normalize function for each interval.

Reply:
Should "normalizeDataPoints" function have "interval" input, isn't it the same then as parent function "setXAxis = (array: number[], interval: IntervalEnum)", should I just separate setXAxis to another file with all cases?

Reply:
I would keep setXAxis because it's related to component, in other graphs you might setYAxis with normalized points :stuck_out_tongue: